src/cmd/run.js

```diff
@@ -107,33 +108,33 @@ export function defaultFirefoxClient(
     maxRetries=250, retryInterval=120,
   }: CreateFirefoxClientParams = {}
 ): Promise<RemoteFirefox> {
-  var retries = 0;
-
-  function establishConnection() {
-    return new Promise((resolve, reject) => {
-      connectToFirefox()
-        .then((connectedClient) => {
-          log.debug('Connected to the Firefox debugger');
-          resolve(connectedClient);
-        })
-        .catch(onlyErrorsWithCode('ECONNREFUSED', (error) => {
-          if (retries >= maxRetries) {
-            log.debug('Connect to Firefox debugger: too many retries');
-            throw error;
-          } else {
-            setTimeout(() => {
-              retries ++;
-              log.debug(
-                `Retrying Firefox (${retries}); connection error: ${error}`);
-              resolve(establishConnection());
-            }, retryInterval);
-          }
-        }))
-        .catch((error) => {
+  async function establishConnection() {
+    var retries = 0;
+    var lastError;
+
+    while (retries <= maxRetries) {
```

Comment:
@kumar303 Actually, we can achieve it using a generator, e.g. something like:

```
function* range(max) {
  for (let i = 0; i < max; i++) {
    yield i;
  }
}

for (let el of range(3)) {
  console.log(el);
}
```

I rewrote it as a for-loop over a custom generator in cdcbc92, if we like it I can move the helper into a separate file and add a test to cover it, or I can rewrite it directly using the for loop (which is basically there, only hidden inside the generator, which is nice because it lets us to use `const` for the `retries` identifier for the for loop).